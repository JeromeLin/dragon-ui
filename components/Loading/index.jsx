import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Loading extends Component {
  render() {
    const { visible, children, style } = this.props;

    return visible ? (
      <div className="ui-loading" style={style}>
        <div className="ui-loading-spins">
          <span className="ui-loading-spin ui-loading-spin-first" />
          <span className="ui-loading-spin ui-loading-spin-second" />
          <span className="ui-loading-spin ui-loading-spin-third" />
        </div>
        <div className="ui-loading-inner">{children}</div>
      </div>
    ) : (
      children
    );
  }
}

Loading.propTypes = {
  visible: PropTypes.bool,
};

Loading.defaultProps = {
  visible: false,
};

export default Loading;
