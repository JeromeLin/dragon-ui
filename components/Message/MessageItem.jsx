import React, { Component } from 'react';
import classnames from 'classnames';
import Icon from '../Icon';

class MessageItem extends Component {
  constructor(props) {
    super(props);
    this.state = {
      visible: false
    };
  }

  componentDidMount() {
    setTimeout(() => this.setState({ visible: true }), 50);
  }

  render() {
    const { content, theme, style } = this.props;

    let wrapClass = classnames({
      'ui-message-wrapper': true,
      'ui-message-visible': this.state.visible
    });

    let iconClass = classnames({
      [`ui-message-${theme}`]: !!theme
    });

    let iconType = 'info-round';

    switch (theme) {
      case 'warning':
        iconType = 'warning-round';
        break;
      case 'success':
        iconType = 'right-round';
        break;
      case 'error':
        iconType = 'wrong-round';
        break;
      default:
        break;
    }

    return (
      <div className={wrapClass} style={style}>
        <div className="ui-message-container">
          <Icon
            type={iconType}
            className={iconClass}
            style={{ fontSize: 15 }}
          />
          {content}
        </div>
      </div>
    );
  }
}

export default MessageItem;
