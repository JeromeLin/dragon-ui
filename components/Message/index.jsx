
import React, { Component, PropTypes } from 'react';
import classnames from 'classnames';

class Message extends Component {

  constructor(props) {
    super(props);
    this.state = {
      timer: undefined,
      visible: false
    };
  }

  componentDidUpdate(){
    clearTimeout(this.state.timer);
    console.log(this.props.message)
    if( this.props.message.length )
      this.state.timer = setTimeout(() => {

        this._onClose();
      }, this.props.duration)
  }

  _onClose(){
    this.props.message.shift();
    this.setState({
      visible: false
    });
  }

  render () {
    const { message, duration, ...others } = this.props;

    let classname = classnames({
      'ui-message-wrapper': true,
      'ui-message-visible': true
    })

    return <div className="ui-message" {...others}>
      {message.map((o,i) => <div key={i} className={classname}>
        <div className="ui-message-container">
          内容:{o} - 时间:{duration}
        </div>
      </div>)}
    </div>

  }

  _onContainerClick(e) {
    e.stopPropagation();
  }

}

Message.propTypes = {
  message     : PropTypes.array,
  duration    : PropTypes.number,
};

Message.defaultProps = {
  message     : [],
  duration    : 1500
};

export default Message;

