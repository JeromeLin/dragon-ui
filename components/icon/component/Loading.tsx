import React from 'react';

const SvgLoading = props => (
  <svg
    viewBox="0 0 31 32"
    fill="currentColor"
    stroke="currentColor"
    width="1em"
    height="1em"
    {...props}
  >
    <path
      d="M15.013 3.847a2.518 2.518 0 1 0-4.277-2.659 2.518 2.518 0 0 0 4.277 2.659M9.79 5.422a2.379 2.379 0 1 0-4.72-.597 2.379 2.379 0 0 0 4.72.597M5.842 8.965a2.239 2.239 0 1 0-4.295 1.265 2.239 2.239 0 0 0 4.295-1.265m-2.141 4.92a2.099 2.099 0 1 0-3.204 2.711A2.099 2.099 0 0 0 3.7 13.885m.188 5.623a1.96 1.96 0 1 0-1.66 3.553 1.96 1.96 0 0 0 1.66-3.553m2.507 4.877a1.822 1.822 0 1 0-.063 3.643 1.822 1.822 0 0 0 .063-3.643m4.274 3.335a1.683 1.683 0 1 0 1.277 3.115 1.683 1.683 0 0 0-1.277-3.115m5.401 1.272a1.544 1.544 0 1 0 2.228 2.14 1.544 1.544 0 0 0-2.227-2.14m5.631-1.218a1.404 1.404 0 1 0 2.648.943 1.405 1.405 0 0 0-2.648-.943m4.413-3.379a1.268 1.268 0 1 0 2.531-.157 1.268 1.268 0 0 0-2.531.157m2.608-4.912a1.13 1.13 0 1 0 2.029-.998 1.13 1.13 0 0 0-2.03.998m.294-5.777"
      fill="currentColor"
      stroke="none"
      fillRule="evenodd"
    />
  </svg>
);

export default SvgLoading;
