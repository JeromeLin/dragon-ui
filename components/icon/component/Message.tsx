import React from 'react';

const SvgMessage = props => (
  <svg
    viewBox="0 0 32 27"
    fill="currentColor"
    stroke="currentColor"
    width="1em"
    height="1em"
    {...props}
  >
    <g stroke="currentColor" strokeWidth={2.6} fill="none" fillRule="evenodd">
      <path d="M27.917 24.846H4a2 2 0 0 1-1.999-2V4a2 2 0 0 1 2-2h23.917a2 2 0 0 1 2 2v18.846a2 2 0 0 1-2 2z" />
      <path d="M2 7.605l13.958 5.818 13.958-5.818" />
    </g>
  </svg>
);

export default SvgMessage;
