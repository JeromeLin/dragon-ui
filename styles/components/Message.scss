$border-radius-base: 6px;
$border-color-base: #d9d9d9;
$box-shadow-base: 0 0 4px rgba(0,0,0,0.17);
$success-color: black;
$error-color: black;
$warning-color: black;
$primary-color: black;

.ui-message {
  font-size: 12px;
  position: fixed;
  z-index: 1050;
  width: 100%;
  top: 16px;
  left: 0;

  .ui-message-wrapper {
    width: auto;
    vertical-align: middle;
    position: absolute;
    left: 50%;
    opacity: 0;
    transform: translate(0, -50px);
    transition: all .5s;
  }
  .ui-message-visible{
    opacity: 1;
    transform: translate(0, 0);
  }
  
  .ui-message-container {
    position: relative;
    right: 50%;
    padding: 8px 16px;
    border-radius: $border-radius-base;
    border: 1px solid $border-color-base;
    box-shadow: $box-shadow-base;
    background: #fff;
    display: block;
  }

  .ui-message-success.ui-icon {
    color: $success-color;
  }

  .ui-message-error.ui-icon {
    color: $error-color;
  }

  .ui-message-warn.ui-icon {
    color: $warning-color;
  }

  .ui-message-info.ui-icon,
  .ui-message-loading.ui-icon {
    color: $primary-color;
  }

  .ui-icon {
    margin-right: 8px;
    font-size: 14px;
    top: 1px;
    position: relative;
  }

}
