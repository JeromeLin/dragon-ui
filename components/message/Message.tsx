import React from "react";
import classnames from 'classnames';
import Icon from '../icon';
import Loading from '../loading';
import { ItemPropsType, IconType } from '../notification/PropsType';
import { mapToIconType, mapToIconTheme } from '../notification/utils';

export default class Message extends React.Component<ItemPropsType, {}> {
  public static defaultProps: ItemPropsType = {
    prefixCls: 'zw-message'
  }
  render() {
    const {
      style, className, prefixCls, icon, message,
      onMouseEnter, onMouseLeave, onClick
    } = this.props;
    let iconToRender = icon ? getIcon(icon, `${prefixCls}__icon`) : null;
    let cls = classnames(`${prefixCls}__content`, { "has-icon": iconToRender });
    return (
      <div className={classnames(prefixCls, className)}
        onMouseEnter={onMouseEnter}
        onMouseLeave={onMouseLeave}
        style={style}
      >
        <div className={cls} onClick={onClick}>
          {iconToRender}
          <div>{message}</div>
        </div>
      </div>
    )
  }
}

function getIcon(icon: IconType | React.ReactElement, className: string) {
  if (React.isValidElement(icon)) {
    return (
      <div className={className}>{icon}</div>
    )
  }
  else if (icon === 'loading') {
    return (
      <Loading className={className} visible size="sm" />
    )
  }
  const iconType = mapToIconType(icon);
  return iconType ?
    <Icon type={mapToIconType(icon)}
      className={className}
      size="sm"
      theme={mapToIconTheme(icon)} />
    : null
}
