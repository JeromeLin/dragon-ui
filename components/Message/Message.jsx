import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MessageItem from './MessageItem';

class Message extends Component {
  constructor(props) {
    super(props);
    this.state = {
      timer: undefined,
      active: 0
    };
  }

  componentWillReceiveProps(nextProps) {
    clearTimeout(this.state.timer);
    this.enter();
  }

  enter() {
    this.state.timer = setTimeout(() => {
      let node = this.message.children;

      if (!node[this.state.active]) return;

      node[this.state.active].className = 'ui-message-wrapper';

      this.state.active++;

      if (this.state.active !== this.message.children.length) this.enter();
    }, this.props.duration);
  }

  render() {
    const {
      msg, duration, theme, style
    } = this.props;

    // eslint-disable-next-line
    let items = msg.map((o, i) => {
      return (
        // eslint-disable-next-line
        <MessageItem key={i} content={o.m} duration={duration} theme={theme} />
      );
    });

    return (
      <div
        className="ui-message"
        ref={(message) => { this.message = message; }}
        style={style}
      >
        {items}
      </div>
    );
  }
}

Message.propTypes = {
  msg: PropTypes.arrayOf(PropTypes.object),
  duration: PropTypes.number,
  theme: PropTypes.oneOf(['info', 'warning', 'success', 'error'])
};

Message.defaultProps = {
  msg: [],
  duration: 1500,
  theme: 'info'
};

export default Message;
