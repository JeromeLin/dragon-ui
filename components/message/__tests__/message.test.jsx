import React from "react";
import { render, shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import Message from "../Message";
import Loading from "../../loading";

describe("Message", () => {
  const message = "This is a test messgae";
  const prefixCls = Message.defaultProps.prefixCls;

  it("renders basic Message correctly", () => {
    const wrapper = shallow(<Message message={message}></Message>);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.text().indexOf(message)).not.toBe(-1);
  });

  it("trigger event correctly", () => {
    const mouseEnter = jest.fn();
    const mouseLeave = jest.fn();
    const click = jest.fn();
    const wrapper = mount(
      <Message
        message={message}
        onMouseEnter={mouseEnter}
        onMouseLeave={mouseLeave}
        onClick={click}
      ></Message>
    );
    wrapper
      .find("div")
      .at(0)
      .simulate("mouseenter")
      .simulate("mouseleave");
    wrapper.find("." + prefixCls + "__content").simulate("click");
    expect(mouseEnter).toHaveBeenCalledTimes(1);
    expect(mouseLeave).toHaveBeenCalledTimes(1);
    expect(click).toHaveBeenCalledTimes(1);
  });

  it("render custom icon correctly", () => {
    const wrapper = mount(
      <Message message={message} icon={<i>?</i>}></Message>
    );
    expect(
      wrapper
        .find("." + prefixCls + "__icon")
        .find("i")
        .text()
    ).toBe("?");
  });

  it("render loading correctly", () => {
    const wrapper = mount(<Message message={message} icon="loading"></Message>);
    expect(wrapper.find("." + Loading.defaultProps.prefixCls)).toHaveLength(1);
  });

  it("render unknow icon correctly", () => {
    const wrapper = mount(
      <Message message={message} icon="unknown-icon"></Message>
    );
    expect(wrapper.find("." + prefixCls + "__icon")).toHaveLength(0);
  });
});
