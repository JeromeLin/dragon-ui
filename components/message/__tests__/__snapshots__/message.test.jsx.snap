// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message renders basic Message correctly 1`] = `
<div
  className="zw-message"
>
  <div
    className="zw-message__content"
  >
    <div>
      This is a test messgae
    </div>
  </div>
</div>
`;
